Auth/Auth/AbstractAuthService.cs

```diff
@@ -0,0 +1,65 @@
+﻿using System;
+using System.Threading.Tasks;
+using Aerogear.Mobile.Auth.User;
+using AeroGear.Mobile.Auth.Authenticator;
+using AeroGear.Mobile.Auth.Config;
+using AeroGear.Mobile.Auth.Credentials;
+using AeroGear.Mobile.Core;
+using AeroGear.Mobile.Core.Configuration;
+
+namespace AeroGear.Mobile.Auth
+{
+    /// <summary>
+    /// Abstract implementation of <see cref="IAuthService"/>.
+    /// </summary>
+    public abstract class AbstractAuthService : IAuthService
+    {
+        protected CredentialManager CredentialManager { get; set; }
+        protected KeycloakConfig KeycloakConfig { get; }
+        protected IAuthenticator Authenticator { get; set; }
+        public string Type => "keycloak";
+        public bool RequiresConfiguration => true;
+
+        public abstract User CurrentUser();
+
+        /// <summary>
+        /// Initializes a new instance of the <see cref="T:AeroGear.Mobile.Auth.AbstractAuthService"/> class.
+        /// </summary>
+        /// <param name="mobileCore">Mobile core.</param>
+        /// <param name="authenticationConfig">Authentication config.</param>
+        public AbstractAuthService(MobileCore mobileCore, AuthenticationConfig authenticationConfig)
+        {
+            var serviceConfiguration = mobileCore.GetServiceConfiguration(Type);
+            KeycloakConfig = new KeycloakConfig(serviceConfiguration);
+        }
+
+        /// <summary>
+        /// Initiate an authentication flow.
+        /// </summary>
+        /// <returns>The authenticate.</returns>
+        /// <param name="authenticateOptions">Authenticate options.</param>
+        public Task<User> Authenticate(IAuthenticateOptions authenticateOptions)
+        {
+            return Authenticator.Authenticate(authenticateOptions);
+        }
+        
+        public abstract void Configure(MobileCore core, ServiceConfiguration serviceConfiguration);
```

Comment:
I guess this should be removed if we're just going by constructor now?

Reply:
yeah, let's move it to the constructor